inst/tests/tests.Rraw

```diff
@@ -17600,3 +17600,18 @@ test(2186, DT[, if (TRUE) .(a=1L) else .(a=1L, b=2L)], DT,
 DT = setDT(replicate(getOption('width'), 1, simplify = FALSE))
 test(2187, {print(DT, col.names='none'); TRUE}, notOutput="V")
 
+# print shows the total number of rows when exceeding the max.print lines, PR#4778
+old = options(max.print=5L)
+DT = as.data.table(iris)
+test(2188.1, print(DT), output="150 rows in total")
+options(old)
+# ensure invalid multi-byte strings can be printed correctly
+latin1 = "fa\xE7ile"
+Encoding(latin1) = "latin1"
+utf8 = iconv(latin1, "latin1", "UTF-8")
+text = c(latin1, utf8, "aaaaaaaa")
+Encoding(text)[1L] = c('UTF-8')
+test(2188.2, print(data.table(A = text)), output=c(
+  "           A", "1: fa\\xe7ile", "2:    fa\u00e7ile", "3:  aaaaaaaa")
```

Comment:
@shrektan This seems to passing on Linux no problem, but not Windows (appveyor CI fail). I'm not quite following it; i.e. the difference between `\\x` (not `\x`) and `\\u`. Could you add some comments as to the intention of the test, and make it work on Windows please?
Also, I follow 2188.1 but why is 2188.2 part of the same test; i.e. I'm not sure what multi-byte has to do with this footer bug in the number of rows omitted. Is it because `cat_matrix` is a new big function so you wanted to test `cat_matrix`?  I would have hoped we have enough tests on `print()` multi-byte already, but happy to see more.  Just an extra comment would be good explaining why 2188.2 is together with 2188.1 when they are seemingly unrelated please.